File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> pascals;
        for(int i=1;i<=numRows;i++){
            vector<int> row;
            if(i==1){
                row.emplace_back(1);
            }else if(i==2){
                row.emplace_back(1);
                row.emplace_back(1);
            }else{
                vector<int> prevRow = pascals[i-2];
                int l = prevRow.size();
                row.emplace_back(1);
                for(int j = 0;j<l-1;j++){
                    row.emplace_back(prevRow[j] + prevRow[j+1]);
                }
                row.emplace_back(1);
            }
            pascals.emplace_back(row);
        }
        return pascals;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> tri = new ArrayList<>();
        for(int i = 0;i<numRows;i++){
            if(i==0){
                List<Integer> hi = new ArrayList<>();
                hi.add(1);
                tri.add(hi);
            }else{
                List<Integer> newList = new ArrayList<>();
                List<Integer> prevList = new ArrayList<>(tri.get(tri.size()-1));
                newList.add(1);
                for(int j = 0;j<prevList.size()-1;j++){
                    newList.add(prevList.get(j)+prevList.get(j+1));
                }
                newList.add(1);
                tri.add(newList);
            }
        }
        return tri;
    }
}